### PR TITLE
VAPE NATION

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
add: VAPE NATION IS COMING TO A RESEARCH AND DEVELOPMENT LAB FITTED WITH MAGNETS 2 NEAR YOU TODAY! CHECK OUT THE FREE UNDERSHIRT TOO!
/:cl:

dreamseeker_2016-07-15_06-04-16

### CODE REVIEW

**Comment by InsaneHyena** [comment]
We get it, you vape.

File: code/modules/research/designs/medical_designs.dm [hunk 1]
```diff
@@ -115,6 +115,17 @@
 	build_path = /obj/item/device/extinguisher_refill
 	category = list("Medical Designs")
 
+/datum/design/ecig
+	name = "Electronic Cigarette"
+	desc = "nobody can read these"
```
Comment (GunHog) [inline code/modules/research/designs/medical_designs.dm]: One day I will change the protolathe so we can, and I will moderately annoyed for having to fix your descriptions!

**Comment by Stealthkibbler** [comment]
+1

**Comment by Chocobro** [comment]
:-1:
colour (note: no s) on that shirt is terrible. misshapen, looks separate from the shirt, incredibly blocky. put some detail into it. 
you're also missing various vape liquids required to make the thing work. get on that. maybe different chems are inhaled, albeit weaker?

**Comment by Iamgoofball** [comment]
@Chocobro its open container pour your own vape liquids

**Comment by Iamgoofball** [comment]
also no spriters wanted to finish it

**Comment by Chocobro** [comment]
it's your sprite. finish it yourself. 
vapes are usually screwed together, so an open container breaks muh realism

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 2]
```diff
@@ -213,26 +214,40 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
-	smoketime--
-	if(smoketime < 1)
-		new type_butt(location)
-		if(ismob(loc))
-			M << "<span class='notice'>Your [name] goes out.</span>"
-			M.unEquip(src, 1)	//un-equip it so the overlays can update //Force the un-equip so the overlays update
-		qdel(src)
-		return
-	open_flame()
+	if(!e_cig)
```
Comment (GunHog) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Why not just omit all this stuff by overriding the proc in `/ecig`?

**Comment by Giacom** [comment]
Oh god.

Yes if you add a phantom cigar.

**Comment by SkowronX** [comment]
meme features!

**Comment by Fox-McCloud** [comment]
Needs to be reagentized so you can fill it with any reagent you want.

**Comment by Bill-Luxe** [comment]
Goof never stop...

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 3]
```diff
@@ -213,26 +214,40 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+		if(isliving(loc))
+			M.IgniteMob()
+		smoketime--
+		if(smoketime < 1)
+			new type_butt(location)
+			if(ismob(loc))
+				M << "<span class='notice'>Your [name] goes out.</span>"
+				M.unEquip(src, 1)	//un-equip it so the overlays can update //Force the un-equip so the overlays update
```
Comment (JJRcop) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Hello, you've reached the Department of Redundancy Department. I'm Agent Mike Michael Michaelson, your Agent for today, how can I help you?

File: code/modules/research/designs/medical_designs.dm (hunk 1 above)
Comment (Core0verload) [inline code/modules/research/designs/medical_designs.dm]: I can.
http://c0.pythonanywhere.com/design

Comment (Core0verload) [inline code/modules/research/designs/medical_designs.dm]: I can.
http://c0.pythonanywhere.com/design

**Comment by Super3222** [comment]
These should spawn with beach bums.
And those beach bums should spawn with vape nation shirts.

**Comment by Lijah99** [comment]
How about no?

**Comment by FlavoredCactus** [comment]
First cannabis, now h3h3 memes.

next goof will make *fart

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/game/objects/items/weapons/cigs_lighters.dm]: then I'd be copypasting

**Comment by InsaneHyena** [comment]
> next goof will make *fart


Goofball, more like Goonball

**Comment by jughu** [comment]
The NEWNEW tactical smoke play

**Comment by Iamgoofball** [comment]
@optimumtact ready for merge

**Comment by Chocobro** [comment]
have you improved the undershirt sprite?
if not, it isn't ready

**Comment by Nienhaus** [comment]
Don't merge yet, I supplied some icons to Goof.
lbphatp
cbgx9kk

**Comment by Bawhoppen** [comment]
Dear lord this PR is attracting ever downstream coder there is

**Comment by optimumtact** [comment]
Goofer better hurry up

**Comment by optimumtact** [comment]
Goof

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 2 above)
Comment (MrStonedOne) [inline code/game/objects/items/weapons/cigs_lighters.dm]: not if you learn the power of ..() and MAGIC

**Comment by FlavoredCactus** [comment]
Let the memes die in peace, oranges.

**Comment by jughu** [comment]
THERE IS STILL HOPE

**Comment by optimumtact** [comment]
no hope

**Pull request closed without merging**